File: src/main/java/com/damianryan/octopus/OctopusProperties.java
package com.damianryan.octopus;

import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;
import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "octopus")
@Data
@FieldDefaults(level = AccessLevel.PRIVATE)
class OctopusProperties {

    String accountUrl;

    String productsUrl;

    String electricityConsumptionUrl;

    String electricityTariffsUrl;

    String gasConsumptionUrl;

    String gasTariffsUrl;

    String greenFixedRateProductsUrl;

    String apiKey;

    String region;

    String tariffType;

    String standingChargesPath;

    String standardUnitRatesPath;

    String fixedRateProductCode;

    String goProductCode;
}

File: src/main/kotlin/com/damianryan/octopus/OctopusProperties.kt
package com.damianryan.octopus

import java.time.ZonedDateTime
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.bind.ConstructorBinding

/**
 * Octopus application properties.
 *
 * @property accountNumber account number
 * @property apiKey API key
 * @property arrangementDate date when arrangement was made with Octopus
 * @property periodFrom date when switch to Octopus actually started
 * @property baseUrl Octopus API base URL
 */
@ConfigurationProperties(prefix = "octopus")
data class OctopusProperties
@ConstructorBinding
constructor(
    val accountNumber: String = "TBA",
    val apiKey: String = "TBA",
    val arrangementDate: ZonedDateTime,
    val periodFrom: ZonedDateTime,
    val baseUrl: String,
) {
    val accountsUrl = "/accounts/$accountNumber"
}
